flight/PiOS/STM32/pios_servo.c

```diff
@@ -391,7 +414,18 @@ int PIOS_Servo_SetMode(const uint16_t *out_rate, const int banks, const uint16_t
 				case SHOT_DSHOT300:
 				case SHOT_DSHOT600:
 				case SHOT_DSHOT1200:
+#if defined(PIOS_INCLUDE_DMASHOT)
+					if (PIOS_DMAShot_IsConfigured() && PIOS_DMAShot_RegisterServo(chan)) {
```

Comment:
Important test case: PIOS_INCLUDE_DMASHOT on a target with no dma channels configured (To verify fallback logic)